packages/backup/src/Helper_Script_Manager.php

```diff
@@ -0,0 +1,322 @@
+			return false;
+		}
+
+		// Tally the files in the target directory, and reject if there are too many.
+		$glob_path    = trailingslashit( $dir ) . '*';
+		$dir_contents = glob( $glob_path );
+		if ( count( $dir_contents ) > 2 ) {
+			return false;
+		}
+
+		// Check that the only remaining files are a README and index.php generated by this system.
+		$allowed_files = array(
+			'README'    => self::README_LINES[0],
+			'index.php' => self::INDEX_FILE,
+		);
+
+		foreach ( $dir_contents as $path ) {
+			$basename = basename( $path );
+			if ( ! isset( $allowed_files[ $basename ] ) ) {
+				return false;
+			}
+
+			// Verify the file starts with the expected contents.
+			if ( ! self::verify_file_header( $path, $allowed_files[ $basename ] ) ) {
+				return false;
+			}
+
+			if ( ! unlink( $path ) ) {
+				return false;
+			}
+		}
+
+		// If the directory is now empty, delete it.
+		if ( count( glob( $glob_path ) ) === 0 ) {
+			return rmdir( $dir );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Find an appropriate location for a jetpack-temp folder, and create one
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @return WP_Error|string Relative path to temp directory if successful, WP_Error if not.
+	 */
+	private static function create_temp_directory() {
+		foreach ( self::RELATIVE_INSTALL_LOCATIONS as $relative_dir ) {
+			// Check if the install location is writeable.
+			$absolute_dir = trailingslashit( ABSPATH ) . $relative_dir;
+			if ( ! is_writeable( $absolute_dir ) ) {
+				continue;
+			}
+
+			// Create if one doesn't already exist.
+			$temp_dir = trailingslashit( $absolute_dir ) . self::TEMP_DIRECTORY;
+			if ( ! is_dir( $temp_dir ) ) {
+				if ( ! mkdir( $temp_dir ) ) {
+					continue;
+				}
+
+				// Temp directory created. Drop a README and index.php file in there.
+				self::write_supplementary_temp_files( $temp_dir );
+			}
+
+			return trailingslashit( $relative_dir ) . self::TEMP_DIRECTORY;
+		}
+
+		return new \WP_Error( 'temp_directory', 'Failed to create jetpack-temp directory' );
+	}
+
+	/**
+	 * Write out an index.php file and a README file for a new jetpack-temp directory.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param string $dir Path to Helper Script directory.
+	 */
+	private static function write_supplementary_temp_files( $dir ) {
+		$readme_path = trailingslashit( $dir ) . 'README';
+		self::put_contents( $readme_path, implode( "\n\n", self::README_LINES ) );
+
+		$index_path = trailingslashit( $dir ) . 'index.php';
+		self::put_contents( $index_path, self::INDEX_FILE );
+	}
+
+	/**
+	 * Write a file to the specified location with the specified contents.
+	 *
+	 * @access private
+	 * @static
+	 *
+	 * @param string $file_path Path to write to.
+	 * @param string $contents  File contents to write.
+	 * @return boolean          True if successfully written.
+	 */
+	private static function put_contents( $file_path, $contents ) {
+		global $wp_filesystem;
+
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
```

Comment:
Should we prefix this one with `\\` as it's in the global namespace?

Reply:
Good point. Added. Thanks :)